crates/evm/execution-types/src/execution_outcome.rs

```diff
@@ -9,30 +9,40 @@ use revm::{
 };
 use std::collections::HashMap;
 
+/// The execution outcome of a single block.
+type SingleBlockExecutionOutcome = ExecutionOutcome<Vec<Receipt>, Vec<Request>, u64>;
+
+/// The execution outcome of multiple blocks aggregated together.
+///
+/// Outer receipt vector stores receipts for each block sequentially.
+/// The inner vector stores receipts ordered by transaction number.
+///
+/// Outer requests vector stores requests for each block sequentially.
+/// The inner vector stores requests ordered by transaction number.
+type MultiBlockExecutionOutcome = ExecutionOutcome<Receipts, Requests, Vec<u64>>;
+
 /// Represents the outcome of block execution, including post-execution changes and reverts.
 ///
 /// The `ExecutionOutcome` structure aggregates the state changes over an arbitrary number of
 /// blocks, capturing the resulting state, receipts, and requests following the execution.
 #[derive(Default, Debug, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-pub struct ExecutionOutcome {
+pub struct ExecutionOutcome<Receipts, Requests, Gas> {
```

Comment:
hmm, not a fan of this.

but I also don't have a good solution to this problem atm...

Reply:
I also don't think it's the best but couldn't figure anything else :/ Will just close this for now